src/node_buffer.cc

```diff
@@ -565,6 +566,53 @@ void StringSlice(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(ret);
 }
 
+// Convert the input into an encoded string
+void DecodeUTF8(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);  // list, flags
+
+  if (!(args[0]->IsArrayBuffer() || args[0]->IsSharedArrayBuffer() ||
+        args[0]->IsArrayBufferView())) {
+    return node::THROW_ERR_INVALID_ARG_TYPE(
+        env->isolate(),
+        "The \"list\" argument must be an instance of SharedArrayBuffer, "
+        "ArrayBuffer or ArrayBufferView.");
+  }
+
+  ArrayBufferViewContents<char> buffer(args[0]);
+
+  CHECK(args[1]->IsBoolean());
+  bool ignore_bom = args[1]->BooleanValue(env->isolate());
+
+  const char* data = buffer.data();
+  size_t beginning = 0;
+  size_t length = buffer.length();
+
+  if (length == 0) return args.GetReturnValue().SetEmptyString();
+
+  if (!ignore_bom && length >= 3) {
+    char bom[] = "\xEF\xBB\xBF";
+
+    if (memcmp(data, bom, 3) == 0) {
```

Comment:
```suggestion
    if (memcmp(data, "\xEF\xBB\xBF", 3) == 0) {
```
`char bom[] = ...` creates an on-stack variable, then copies the string into it. In theory, the compiler can optimize that away but compilers are fickle beasts.

Reply:
Updated pull request. Thank you for the review.

Reply:
> In theory, the compiler can optimize that away but compilers are fickle beasts.

With the length being known at compile time (and being small enough to fit into a register) and builtin optimizations enabled (which gets enabled at `-O2`) - both gcc and clang will easily optimize the entire function call away: https://godbolt.org/z/7aYhWdzW7